packages/cozy-pouch-link/src/CozyPouchLink.js

```diff
@@ -83,80 +53,44 @@ export default class PouchLink extends CozyLink {
       )
     }
     this.doctypes = doctypes
-    this.pouches = createPouches(this.doctypes)
     this.indexes = {}
   }
 
-  registerClient(client) {
+  async registerClient(client) {
     this.client = client
+    if (this.pouches) {
+      await this.pouches.destroy()
+    }
+    this.pouches = new PouchManager(this.doctypes, {
+      getReplicationURL: doctype =>
+        getReplicationURL(this.client.uri, this.client.token, doctype),
+      onError: err => this.onSyncError(err)
+    })
     if (client && this.options.initialSync) {
-      this.syncAll()
+      this.pouches.startReplications()
     }
   }
 
   async reset() {
-    await this.resetAllDBs()
+    await this.pouches.destroy()
     this.client = undefined
   }
 
-  getAllDBs() {
-    return Object.values(this.pouches).map(x => x.db)
-  }
-
-  resetAllDBs() {
-    return Promise.all(this.getAllDBs().map(db => db.destroy()))
-  }
-
-  getDBInfo(doctype) {
-    const dbInfo = this.pouches[doctype]
-    if (!dbInfo) {
-      throw new Error(`${doctype} not supported by cozy-pouch-link instance`)
-    }
-    return dbInfo
-  }
-
-  getDB(doctype) {
-    return this.getDBInfo(doctype).db
-  }
-
-  syncOne(doctype) {
-    return new Promise((resolve, reject) => {
-      const info = this.getDBInfo(doctype)
-      if (info.syncing) {
-        return resolve(info.syncing)
-      } else {
-        const replication = info.db.replicate.from(
-          this.getReplicationUrl(doctype),
-          {
-            batch_size: 1000 // we have mostly small documents
-          }
-        )
-        replication.on('error', err => {
-          console.warn('Error while syncing', err)
-          reject('Error while syncing')
-        })
-        info.syncing = replication.on('complete', result => {
-          info.syncing = null
-          resolve(result)
-        })
-      }
-    })
-  }
-
-  syncAll() {
-    return Promise.all(
-      this.doctypes.map(doctype => this.syncOne(doctype))
-    ).then(pipe(() => this.onSync()))
-  }
-
   onSync() {
     this.synced = true
   }
 
-  getReplicationUrl(doctype) {
-    const client = this.client
-    const basicAuth = client.token.toBasicAuth()
-    return (client.uri + '/data/' + doctype).replace('//', `//${basicAuth}`)
+  async onSyncError(error) {
+    if (isExpiredTokenError(error)) {
+      try {
+        await this.client.renewAuthorization()
+        this.pouches.startReplicationLoop()
+      } catch (err) {
+        console.warn('Could not refresh token, replication has stopped', err)
```

Comment:
Are we sure we don't want to throw the exception anyway?

Reply:
Yes. It is an error but it should not break the application.

There will be code that will try to restart the replication when mobile gets the focus so I think it should not that much of a problem.